### DISCUSSION TITLE
Question about Scraping Messages from an Open Telegram Channel and Saving to XML using this Library

### DISCUSSION THREAD

**Discussion by YuliyaMinsk** [yagop/node-telegram-bot-api#1133] [Q&A]
Hello everyone,

I'm wondering if it's possible to create a Telegram bot using this JavaScript library that can scrape messages from a selected Telegram channel and save them into an XML file. The catch is, I don't have admin access to the channel, but the channel itself is public.

I've heard that this kind of functionality is only possible using Python. Is that true? Can this library handle such a use-case?

Would love to hear your thoughts and suggestions.

Thanks in advance!

**Comment by FebX23** [comment]
For sure, you can totally build a bot that grabs messages from a public Telegram channel and squirrels them away into an XML file, and you don't need Python for that; JavaScript's got your back. The whole "_Python is a must_" vibe is more about preference and what people are used to rather than a hard rule.

The trick is to pick a JS library that plays nice with Telegram's API, especially one that lets you listen in on public channels. Once you've got the messages, turning them into XML is no sweat in JavaScript.

Since the channel's public, not having admin access isn't a roadblock. You're good to go as long as you stick to the rules Telegram sets out.

If you're worried about running into any scrapes with rate limits or getting your IP frowned upon, [Crawlbase](https://crawlbase.com/?utm_source=GitHub_Feb) could be a handy sidekick. It's smart about handling web requests, making your life a bit easier. Just, you know, keep it on the level and respect usage policies.
